trainers/build_trainers.py

```diff
@@ -35,8 +35,13 @@ def ddp_setup(rank, world_size):
         rank: Unique identifier of each process
         world_size: Total number of processes
     """
-    os.environ["MASTER_ADDR"] = "localhost"
-    os.environ["MASTER_PORT"] = "12355"
+    # Get the master address and port from SLURM environment variables
+    master_addr = os.environ.get('MASTER_ADDR', 'localhost')
```

Comment:
this allows us to use multi multigpu runs at a time

Reply:
yea this is better. It's more dynamic